junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertionsTests.kt

```diff
@@ -96,32 +96,26 @@ class KotlinAssertionsTests {
         dynamicContainer("fails when an exception is thrown", Stream.of(
             dynamicTest("for no arguments variant") {
                 val exception = assertThrows<AssertionError> {
-                    assertDoesNotThrow {
+                    assertDoesNotThrow<Unit> {
```

Comment:
Oh and, as-is, I believe this is a breaking change. Reified type parameters does not play nicely with `Nothing`

Reply:
Does this mean current usages of `assertDoesNotThrow` will break?

Reply:
> Does this mean current usages of `assertDoesNotThrow` will break?

Only the ones where the provided `executable` returns a type that is not [compatible with reified type parameters](https://github.com/JetBrains/kotlin/blob/master/spec-docs/reified-type-parameters.md). Currently I believe is only the fictitious `Nothing` type.

> Could this not break if we "artificially" change the return type to Unit? (genuine question, I'm not much of a Kotlin master)

Do you mean by overloading the functions with a variant that takes `() -> Nothing` and returns `Unit`? Should work, I think :D! I'll give it a try!

Reply:
I just read a bit more about `reified` and now I'm confused... why do we even need it, since we don't manipulate the `R` type? 🤔

Reply:
Oh, is it to keep the type of the returned object?

Reply:
You're 100% right, in this case having a reified type does not make any sense. I was just looking at the `assertThrows` signature and blindly copied it.

Reply:
> Do you mean by overloading the functions with a variant that takes `() -> Nothing` and returns `Unit`? Should work, I think :D! I'll give it a try!

To answer this, yes seems possible in later versions (seems to still be experimental):
- https://discuss.kotlinlang.org/t/overloaded-function-with-lambda-parameter-of-different-return-types/6053/4
- https://youtrack.jetbrains.com/issue/KT-11265

Reply:
I removed the reified type parameters in https://github.com/junit-team/junit5/pull/2684/commits/80aeb8da903a46074e756f5c5c23c97b7c2d4e05

Reply:
Ahah, glad I could help! (Can I call myself a kotlin expert now? 😎)